src/IJ_mv/Makefile

```diff
@@ -40,12 +42,20 @@ FILES =\
  IJ_assumed_part.c\
  IJMatrix.c\
  IJMatrix_parcsr.c\
- IJMatrix_parcsr_device.c\
  IJVector.c\
- IJVector_parcsr.c\
+ IJVector_parcsr.c
+
+CUFILES =\
+ IJMatrix_parcsr_device.c\
  IJVector_parcsr_device.c
 
-OBJS = ${FILES:.c=.o}
+COBJS = ${FILES:.c=.o}
+ifeq ($(CUCC), )
```

Comment:
Why not use a rule in Makefile.config.in instead of having this if statement?  The '.obj' rule can be the same as the '.o' rule when not using CUDA.

Reply:
OK. The goal is to have a different rule to compile certain `.c` files to `.obj` when CUDA is enabled, and use the same rule from `.c` to `.o` without CUDA. I don't want to rename certain `.c` files to another extension such as `.cu`. I don't know if I understand what you were suggesting. It is not clear for me how to switch rules in `Makefile.config`.

Reply:
If you always did this
```
CUOBJS = ${CUFILES:.c=.obj}
```
Then, if the Makefile.config rule is as follows, these CUFILES files would be compiled with the `CUCC` compiler
```
.c.obj:
	$(CUCC) $(CUFLAGS) -c $< -o $@
```
On the other had, with this rule, they would be compiled with the `CC` compiler
```
.c.obj:
	$(CC) $(CFLAGS) -c $<
```
Isn't the `CUCC` thing determined at configure time?  If so, we should be able to write an appropriate rule for `.c.obj:`.  I think this makes sense and does not require renaming files, but maybe I'm confused.  I'm just trying to avoid having the same `ifeq` construct repeated in a bunch of makefiles.

Is there a special meaning attached to the `.obj` extension?